content/en/tracing/trace_collection/automatic_instrumentation/compile_time/go.md

```diff
@@ -0,0 +1,135 @@
+
+Orchestrion only supports [go modules][7].
+
+### Framework Support
+
+While the goal is to eventually support all integrations available in the [Go Datadog Trace library][4], support is currently only built for a gowing subset of them. The complete list of supported frameworks can be found in the [documentation][8].
+
+## Getting Started
+
+### Installation
+
+1. Install Orchestrion in your environment:
+   ```sh
+   go install github.com/datadog/orchestrion@latest
+   ```
+2. If necessary, also add `GOBIN` to your `PATH`:
+   ```sh
+   export PATH="${PATH}:$(go env GOBIN)"
+   ```
+3. Register Orchestrion in your project's `go.mod` to ensure reproductibe builds:
+   ```sh
+   orchestrion pin
+   ```
+4. Commit changes to your version control system:
+   ```sh
+   git commit -m "chore: enable orchestrion" go.mod go.sum orchestrion.tool.go
+   ```
+
+### Usage
+
+Adjust your build commands or environment to be Orchestrion-enabled using either of these options:
+
+- **Use `orchestrion go`:** Simply prepend all your usual `go` commands with `orchestrion`. For example:
+  ```sh
+  orchestrion go build .
+  orchestrion go run .
+  orchestrion go test ./...
+  ```
+- **Add a `-toolexec` argument:** Manually add the required `-toolexec="orchestrion toolexec"` argument to your `go` commands:
+  ```sh
+  go build -toolexec="orchestrion toolexec" .
+  go run -toolexec="orchestrion toolexec" .
+  go test -toolexec="orchestrion toolexec" ./...
+  ```
+- **Modify `$GOFLAGS`:** Modify the `$GOFLAGS` environment variable to inject orchestrion, and use `go` commands normally:
+  ```sh
+  # Make sure to include the quotes as shown below, as these are required for
+  # the Go toolchain to parse GOFLAGS properly!
+  export GOFLAGS="${GOFLAGS} '-toolexec=orchestrion toolexec'"
+  go build .
+  go run .
+  go test ./...
+  ```
+
+## Manual instrumentation
+
+While Orchestrion automatically instruments your application for supported [integrations][4], there may be cases where
+you still want to manuall add instrumentation to specific sections of your code.
+
+### Custom trace spans
+
+Apply the `//dd:span` comment on any function that accepts a `context.Context` or `*http.Request` argument, and Orchestrion will automatically create a trace span for all invocations of this function. This also works for function expressions.
+
+```go
+//dd:span custom_tag:tag_value
+func CriticalPathFunction(ctx context.Context) {
+  // ... implementation details ...
+}
+```
+
+### Using the Tracer library
+
+You can use the [Tracer library][4] in your application even if Orchestrion is in use. This can be used to instrument features that Orchestrion does not currently support.
+
+If Orchestrion get support of this feature at a later point, the manual call may result in duplicated spans being present in the traces, which cane asily be resolved by removing the manual calls that are no longer needed. Refer to Orchestrion [release notes][9] to know about new features.
```

Comment:
what about rather documenting right now to put manual instrum in separate files tagged with `dd:ignore` so you get something future-proof and safer directly?

Reply:
You can't tag a whole file, only a given AST node... unless we allow it on the `package` declaration (which is not a thing today). Also, that works for your own code, but not your dependencies (if these are already instrumented for Datadog)